geode-core/src/main/java/org/apache/geode/internal/cache/TXLastEventInTransactionUtils.java

```diff
@@ -77,11 +78,7 @@ private static boolean checkNoSendersGroupTransactionEvents(List<EntryEventImpl>
       Cache cache) throws ServiceConfigurationError {
     for (String senderId : getSenderIdsForEvents(callbacks)) {
       GatewaySender sender = cache.getGatewaySender(senderId);
-      if (sender == null) {
-        logger.error("No sender found for {}", senderId);
-        throw new ServiceConfigurationError("No information for senderId: " + senderId);
-      }
-      if (sender.mustGroupTransactionEvents()) {
+      if (sender != null && sender.mustGroupTransactionEvents()) {
```

Comment:
The "ServiceConfigurationError" is thrown in "getLastTransactionEvent" that can also be removed by returning null; as done here...

Reply:
In the flow of the code to getLastTransactionEvent, checkNoSendersGroupTransactionEvents was executed before doesSenderGroupTransactionEvents.

Method checkNoSendersGroupTransactionEvents returns false if there is one op needs to send to a GatewaySender which has mustGroupTransactionEvents set to true. Otherwise doesSenderGroupTransactionEvents method won't be executed (getLastTransactionEvent returns null in that case).

If in a tx, there is one op needs to send to gateway with mustGroupTransactionEvents setting, all ops in the transaction should as well. If not, it means user did not configure the senders correctly (for the transaction executed).